pep-0704.rst

```diff
@@ -0,0 +1,137 @@
+
+This PEP proposes that the default behaviour of installers like ``pip`` should be to require a virtual environment to be active. This would make it easier for new users to get started with Python, and would reduce the scope for mistakes.
+
+
+Specification
+=============
+
+When a user runs an installer without an active virtual environment, the installer MUST print an error message and exit with a non-zero exit code. The error message MUST inform the user that a virtual environment is required, SHOULD provide shell-specific instructions on how to create and activate a virtual environment named ``.venv``, and SHOULD provide a link to a documentation page that explains how to create and activate a virtual environment.
+
+The installer SHOULD also provide a mechanism to disable this behaviour, allowing a user to opt-in to running the tool outside of a virtual environment. The installer MAY include this in the error message that it prints, and MUST include it in the documentation page it links to.
+
+If the installer includes a link to a documentation page in the error message described above, the documentation page SHOULD be written with a focus on new users. It should explain what virtual environments are, why they are required, and how to create and activate a virtual environment. It SHOULD include instructions for the most common shells and platforms.
+
+Installers MAY choose to implement this default behaviour on any Python versions, but MUST implement it on Python 3.13 or newer.
+
+
+Backwards Compatibility
+=======================
+
+This PEP is backwards incompatible with workflows where users are using installers outside of virtual environments. Such users will be prompted with an error message and will need to either:
+
+- explicitly opt-in to running the installer outside of a virtual environment, or
+- create and use a virtual environment
+
+Users who are already using virtual environments will not be affected by this change.
+
+Workflow tools (which manage virtual environments for the user, under the hood) should be unaffected, since they should already be using the virtual environment for running the installer.
+
+
+Security Implications
+=====================
+
+This PEP does not introduce any new security implications.
+
+
+How to Teach This
+=================
+
+This PEP does require that new users create and use a virtual environment to get started with using Python packages. This is however a best practice, as demonstrated by the section on "basics of how to install Python packages" in the Python Packaging User Guide, which explains how/what virtual environments are before discussing using ``pip``.[1]_
```

Comment:
```suggestion
This PEP does require that new users create and use a virtual environment to get started with using Python packages. This is, however, a best practice, as demonstrated by the `section on the basics of how to install Python packages <https://packaging.python.org/en/latest/tutorials/installing-packages/#creating-virtual-environments>`__ in the Python Packaging User Guide, which explains what virtual environments are before discussing using ``pip``.
```

* Fixed a couple grammar errors.

* The footnote syntax is broken.

* Further, It doesn't really make sense to require to user to find and click through to a footnote, just to have to click on another (bare) link. Instead, just link the content directly. Per PEP 12:

  > Native reST hyperlink targets or inline hyperlinks in the text should be used in preference to footnotes for including URLs to online resources.

  See also #2130

* In fact, you could normally just do `` :ref:`sections-ref-target-label-on-pypug` `` to link it directly, but unfortunately it doesn't (yet) have one.